packages/cli/src/lib/runPreBuildTasks.js

```diff
@@ -0,0 +1,32 @@
+import path from 'path'
+
+import { getProject } from '@redwoodjs/structure'
+
+const project = getProject()
+
+function generateTypeDef(filename, contents) {
+  const { host } = project
+  host.writeFileSync(path.join(host.paths.types, filename), contents)
```

Comment:
Should we check if the file exists before writing it again?

Reply:
Seems sensible. Wouldn't we want to always have the latest typings though?

Lets say I update that type in a future verison, always writing means users always would have the latest ones generated

Reply:
Maybe we can blow them out when the user upgrades, since these files aren't added to git?

What I was actually trying to avoid was a situation where a user decided to manually type them (for whatever reason).

Reply:
So what I was thinking around this was that if the user wanted to override the types they would add it to one of these folders: `types`, `web/types` or `api/types`

Declaration merging will work the same way, so if they wanted to merge (on top of this) they can. If they wanted to completely override, they could

Reply:
Added types and "../types" to paths in both tsconfig templates